postgresql/database_test.go

```diff
@@ -1912,3 +1913,50 @@ func TestOptionTypeJsonbStruct(t *testing.T) {
 		t.Fatalf("Expecting Num to be 123")
 	}
 }
+
+// TestExhaustConnections simulates a "too many connections" situation
+// triggered by opening more transactions than available connections.
+// upper.io/db deals with this problem by waiting a bit more for the connection
+// to be established.
+func TestExhaustConnections(t *testing.T) {
```

Comment:
This method simulates concurrent transaction queries that drain the PostgreSQL connection limit for a while.